src/main/java/org/spongepowered/api/CatalogType.java

```diff
@@ -60,5 +64,13 @@
      * @return The human-readable name of this catalog type
      */
     String getName();
+    
+    @Override
+    default Text toText() {
```

Comment:
`TextRepresentable` is **not** about `How I want it to be displayed  in chat`.
It is about `How IT want to represent itself in chat`.
Its the best-effort approach if you don't know anything about the internals of the class.

Using a `TextRepresenter`s is something totally different. That would be the equivalent of adding a `Function<Object,Text>` to `Text.of(Object...)` which would make the method useless since it would only be a for each loop.
````java
Arrays.stream(objects).map(textRepresenter).forEach(builder::append);
````
Maybe a `Function<Object,TextBuilder>` would be a better choice of param. But this still wouldn't solve the issue of the ugly `toString()` default if nothing is specified.